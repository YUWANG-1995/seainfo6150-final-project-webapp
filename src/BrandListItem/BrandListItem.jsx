import React from "react";
import styles from "./BrandListItem.module.css";

const BrandListItem = (props) => {
  return (
    <div className={styles.container}>
        <img className={styles.image} src={props.brand.brandImage._url} alt={props.brand.brand} />
        <p className={styles.brand}>{props.brand.brand}</p>
    </div>
  );
}

export default BrandListItem;
