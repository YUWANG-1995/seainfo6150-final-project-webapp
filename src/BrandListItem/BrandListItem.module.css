.container{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.image{
  width: 300px;
  height: 160px;
}

.brand{
  font-family: 'Open Sans', sans-serif;
  font-weight: bold;
  text-align: center;
  font-size: 50px;
  color: #000000;
}
p.brand:hover{
  color:red;
}
