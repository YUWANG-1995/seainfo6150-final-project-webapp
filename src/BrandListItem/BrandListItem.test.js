import React from "react";
import { render } from "@testing-library/react";
import BrandListItem from "./BrandListItem.jsx";

it("renders the BrandListItem Component Correctly", ()=> {
  const test = {
      brand: 'BMW',
      brandImage: {
        _url: "https://4.bp.blogspot.com/-IOuK59xvd-Q/V0yK19RcD8I/AAAAAAAAAJs/4gavimm0yW8av7K9yZSIvNQ5A-NeW2Y-ACLcB/s1600/bmw-logo.jpg"
      }
  }

  const { container } = render(
    <BrandListItem
      brand = {test}
    />
  );
  expect(container).toMatchSnapshot();
})
