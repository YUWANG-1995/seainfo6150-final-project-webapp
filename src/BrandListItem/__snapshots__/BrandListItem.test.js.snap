// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the BrandListItem Component Correctly 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="BMW"
      class="image"
      src="https://4.bp.blogspot.com/-IOuK59xvd-Q/V0yK19RcD8I/AAAAAAAAAJs/4gavimm0yW8av7K9yZSIvNQ5A-NeW2Y-ACLcB/s1600/bmw-logo.jpg"
    />
    <p
      class="brand"
    >
      BMW
    </p>
  </div>
</div>
`;
